src/checkers/inference/InferrableAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,22 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for all annotatedTypeFactory that wish to be used with
+ * Checker-Framework-Inference
+ */
+
+public interface InferrableAnnotatedTypeFactory {
+
+    /**
+     * return the TreeAnnotator from real type annotatedTypeFactory
+     * 
+     */
+    TreeAnnotator getRealInferenceTreeAnnotator(
+            InferenceAnnotatedTypeFactory atypeFactory,
+            InferrableChecker realChecker,
+            AnnotatedTypeFactory realAnnotatedTypeFactory,
```

Comment:
This parameter is not necessary: it should always hold that realAnnotatedTypeFactory == this, as this method will be implemented by that real ATF.